x/clp/keeper/calculations.go

```diff
@@ -286,6 +287,12 @@ func CalculatePoolUnits(oldPoolUnits, nativeAssetBalance, externalAssetBalance,
 		return sdk.ZeroUint(), sdk.ZeroUint(), types.ErrAsymmetricAdd
 	}
 
+	ratioDiff := R.Quo(A).Sub(r.Quo(a)).Abs()
```

Comment:
@timlind I believe the formula should be instead the following: 

```
|(X/Y) - (x/y)| > t 
```

where `X` is the external asset balance, `Y` native asset balance and x and y external and native amount which means it should translate to:

```
ratioDiff := A.Quo(R).Sub(a.Quo(r)).Abs()
```

correct?

Reply:
Updated.